### PR TITLE
feat(admin): update of the Table-Checkbox service

### PR DESCRIPTION
**Pull request by mattjoke**
* updated TableCheckbox Service
* merged isAllSelected and isAllSelected isAllSelectedWithDisabledCheckbox
* changed the call for the service to be used
  in a pipe

### CODE REVIEW

File: apps/admin-gui/src/app/shared/components/blacklist-list/blacklist-list.component.html [hunk 1]
```diff
@@ -19,8 +19,8 @@
           <mat-checkbox
             (change)="$event ? masterToggle() : null"
             [aria-label]="checkboxLabel()"
-            [checked]="selection.hasValue() && isAllSelected()"
-            [indeterminate]="selection.hasValue() && !isAllSelected()"
+            [checked]="selection.hasValue() && (dataSource | isAllSelected: selection.selected.length :canBeSelected)"
+            [indeterminate]="selection.hasValue() && !(dataSource | isAllSelected: selection.selected.length :canBeSelected)"
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/shared/components/blacklist-list/blacklist-list.component.html:23]: Remove no existing method `canBeSelected`. This problem is in many places, so fix it in all HTML files, where this problem occurs, please.
image

```suggestion
            [checked]="selection.hasValue() && (dataSource | isAllSelected: selection.selected.length)"
            [indeterminate]="selection.hasValue() && !(dataSource | isAllSelected: selection.selected.length)"
```

File: libs/perun/pipes/src/lib/is-all-selected.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+@Pipe({
+  name: 'isAllSelected',
+})
+export class IsAllSelectedPipe implements PipeTransform {
+  constructor(private tableCheckboxService: TableCheckbox) {}
+
+  transform(
+    ds: MatTableDataSource<unknown>,
+    numberSelected: number,
+    canBeSelected: (T) => boolean
```
Comment (HejdaJakub) [inline libs/perun/pipes/src/lib/is-all-selected.pipe.ts:14]: Make this param optional, so you can skip it in HTML files, where this method does not exist.
```suggestion
    canBeSelected?: (T) => boolean
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
I don't know why, but replacing methods in HTML with pipes didn't fix the problem. We can discuss it on a daily standup.

File: libs/perun/pipes/src/lib/is-all-selected.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export class IsAllSelectedPipe implements PipeTransform {
+  constructor(private tableCheckboxService: TableCheckbox) {}
+
+  transform(
+    ds: MatTableDataSource<unknown>,
+    numberSelected: number,
+    canBeSelected?: (T) => boolean
+  ): boolean {
+    return this.tableCheckboxService.isAllSelected(numberSelected, ds, canBeSelected);
+  }
```
Comment (bodnara) [inline libs/perun/pipes/src/lib/is-all-selected.pipe.ts:17]:
```suggestion
  transform<T> (
    ds: MatTableDataSource<T>,
    numberSelected: number,
    canBeSelected?: (item: T) => boolean
  ): boolean {
    return this.tableCheckboxService.isAllSelected(numberSelected, ds, canBeSelected);
  }
```

File: libs/perun/services/src/lib/table-checkbox.service.ts [hunk 4]
```diff
@@ -17,26 +17,38 @@ export class TableCheckbox {
...
+  isAllSelected<T>(
     rowsSelected: number,
-    filter: string,
-    pageSize: number,
-    nextPage: boolean,
-    pageIndex: number,
     dataSource: MatTableDataSource<T>,
-    sort: MatSort,
-    canBeSelected: (T) => boolean
+    canBeSelected?: (T) => boolean
```
Comment (bodnara) [inline libs/perun/services/src/lib/table-checkbox.service.ts:23]: If we provide a default select function that just returns true, there is no need for the `if(!canBeSelected)` block I believe.
```suggestion
    canBeSelected: (item: T) => boolean = (): boolean => true
```

**Review by bodnara — CHANGES_REQUESTED** [review]
Some tables have console errors and some do not show properly at all, I did not go through all of them but e.g. the member-list.

Also perhaps it was not specified properly but the masterToggle function can have the number of parameters reduces as well.

**Review by bodnara — APPROVED** [review]
Nice job on this one.

**Review by HejdaJakub — APPROVED** [review]
I have bad news guys because I tested it and the function `isAllSelected()` from the tableCheckbox service is still called many times on every mouse move across the table. Then I found out that we call in these HTML files function `checkboxLabel()`, which calls `isAllSelected()` function and here we go again. But I believe that this part deserves a separate task and a separate PR so that I can approve this PR.

**Pull request merged**